src/mercury_engine_data_structures/formats/bmscc.py

```diff
@@ -14,47 +15,80 @@
 
 from mercury_engine_data_structures import game_check
 from mercury_engine_data_structures.base_resource import BaseResource
-from mercury_engine_data_structures.common_types import StrId, VersionAdapter, make_vector
+from mercury_engine_data_structures.common_types import (
+    StrId,
+    VersionAdapter,
+    make_vector,
+)
 from mercury_engine_data_structures.construct_extensions.misc import ErrorWithMessage
 from mercury_engine_data_structures.formats.collision import collision_formats
-from mercury_engine_data_structures.game_check import Game
+
+if TYPE_CHECKING:
+    from mercury_engine_data_structures.game_check import Game
 
 CollisionEntry = Struct(
-    name=StrId,
-    prop1=StrId,
-    prop2=StrId,
-    prop3=StrId,
-    flag=IfThenElse(
-        game_check.current_game_at_most(Game.SAMUS_RETURNS),
+    "name" / StrId,
+    "prop1" / StrId,
+    "prop2" / StrId,
+    "prop3" / StrId,
+    "flag"
+    / game_check.is_sr_or_else(
         Int8ul,
         Int16ul,
     ),
-    type=StrId,
-    data=Switch(
+    "type" / StrId,
+    "data"
+    / Switch(
         construct.this.type,
         collision_formats,
         ErrorWithMessage(lambda ctx: f"Type {ctx.type} not known, valid types are {list(collision_formats.keys())}."),
     ),
 )
 
 CollisionLayer = Struct(
-    name=StrId,
-    entries=make_vector(CollisionEntry),
+    "name" / StrId,
+    "entries" / make_vector(CollisionEntry),
+)
+
+PartsComponent = Struct(
+    "group" / StrId,
+    "components" / make_vector(Struct("name" / StrId)),
 )
 
 BMSCC = Struct(
-    _magic=Const(b"MSCD"),
-    _version=IfThenElse(
-        game_check.current_game_at_most(Game.SAMUS_RETURNS),
+    "_magic" / Const(b"MSCD"),
+    "_version"
+    / game_check.is_sr_or_else(
         VersionAdapter("1.13.0"),
         VersionAdapter("1.16.0"),
     ),
-    layers=make_vector(CollisionLayer),
-    eof=GreedyBytes,
+    "layers" / make_vector(CollisionLayer),
+    "parts" / construct.Optional(make_vector(PartsComponent)),
+    construct.Terminated,
 )
 
 
 class Bmscc(BaseResource):
     @classmethod
     def construct_class(cls, target_game: Game) -> Construct:
         return BMSCC
+
+    # Bmscc has an entry per collision_camera, Bmscd has one entry per file
+    def get_data(self, layer_idx: int = 0, entry_idx: int = 0) -> Container:
+        return self.raw.layers[layer_idx].entries[entry_idx].data
+
+    # Returns all data associated with a poly (points, boundings)
```

Comment:
each of these comments describing the purpose of a function should be a docstrings instead:
```py
def get_poly(self, poly_idx: int) -> Container:
    """Returns all data associated with a poly (points, boundings)"""
    return self.get_data().polys[poly_idx]
```